output/transform/postgres.go

```diff
@@ -167,6 +170,83 @@ func transformPostgresServerStats(s snapshot.FullSnapshot, newState state.Persis
 		PgStatStatementsReset:             snapshot.NullTimeToNullTimestamp(diffState.PgStatStatementsStats.Reset),
 	}
 
+	for timeKey, diffedStats := range transientState.HistoricServerIoStats {
+		// Ignore any data older than an hour, as a safety measure in case of many
+		// failed full snapshot runs (which don't reset state)
+		if time.Since(timeKey.CollectedAt).Hours() >= 1 {
+			continue
+		}
+
+		h := snapshot.ServerIoStatistics{}
+		h.CollectedAt = timestamppb.New(timeKey.CollectedAt)
+		h.CollectedSecs = timeKey.CollectedIntervalSecs
+
+		for k, stats := range diffedStats {
+			stat := snapshot.ServerIoStatistic{
+				Reads:         stats.Reads,
+				ReadTime:      stats.ReadTime,
+				Writes:        stats.Writes,
+				WriteTime:     stats.WriteTime,
+				Writebacks:    stats.Writebacks,
+				WritebackTime: stats.WritebackTime,
+				Extends:       stats.Extends,
+				ExtendTime:    stats.ExtendTime,
+				OpBytes:       stats.OpBytes,
+				Hits:          stats.Hits,
+				Evictions:     stats.Evictions,
+				Reuses:        stats.Reuses,
+				Fsyncs:        stats.Fsyncs,
+				FsyncTime:     stats.FsyncTime,
+			}
+			switch k.BackendType {
+			case "unknown":
+				stat.BackendType = snapshot.BackendCountStatistic_UNKNOWN_TYPE
+			case "autovacuum launcher":
+				stat.BackendType = snapshot.BackendCountStatistic_AUTOVACUUM_LAUNCHER
+			case "autovacuum worker":
+				stat.BackendType = snapshot.BackendCountStatistic_AUTOVACUUM_WORKER
+			case "background worker":
+				stat.BackendType = snapshot.BackendCountStatistic_BACKGROUND_WORKER
+			case "background writer":
+				stat.BackendType = snapshot.BackendCountStatistic_BACKGROUND_WRITER
+			case "client backend":
+				stat.BackendType = snapshot.BackendCountStatistic_CLIENT_BACKEND
+			case "checkpointer":
+				stat.BackendType = snapshot.BackendCountStatistic_CHECKPOINTER
+			case "startup":
+				stat.BackendType = snapshot.BackendCountStatistic_STARTUP
+			case "walreceiver":
+				stat.BackendType = snapshot.BackendCountStatistic_WALRECEIVER
+			case "walsender":
+				stat.BackendType = snapshot.BackendCountStatistic_WALSENDER
+			case "walwriter":
+				stat.BackendType = snapshot.BackendCountStatistic_WALWRITER
```

Comment:
We're missing the "slotsync worker" backend type here, that was added in Postgres 17: https://www.postgresql.org/docs/current/logicaldecoding-explanation.html#LOGICALDECODING-REPLICATION-SLOTS-SYNCHRONIZATION

(I think to be able to account for all I/O its helpful to include it, even if an edge case)